nevergrad/optimization/optimizerlib.py

```diff
@@ -1875,6 +1875,7 @@ def __init__(self, parametrization: IntOrParameter, budget: tp.Optional[int] = N
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
         descr = self.parametrization.descriptors
         self.has_noise = not (descr.deterministic and descr.deterministic_function)
+        self.small_noise = self.has_noise and descr.deterministic_function  # The noise coming from discrete variables goes to 0.
```

Comment:
any better name? small noise seems really weird and not helping for understanding the code. Maybe not naming it is actually more readable, why not just use has_noise & deterministic_function?

Reply:
I've switched to something explicit, but it's also a part of "small noise", i.e. noise which vanishes around the optimum.
An issue is that typically we rarely know that noise will vanish around the optimum --- except in the present case.